test-common/test-common-api/src/main/java/org/apache/kafka/common/test/api/ClusterInstance.java

```diff
@@ -224,5 +232,46 @@ default void waitAcls(AclBindingFilter filter, Collection<AccessControlEntry> en
             }, "expected acls: " + entries + ", actual acls: " + actualEntries.get());
         }
     }
+    
 
+    //---------------------------[producer/consumer]---------------------------//
+    
+    default <K, V> Producer<K, V> producer(Map<String, Object> overrides, 
+                                           Serializer<K> keySerializer, 
+                                           Serializer<V> valueSerializer
+    ) {
+        Properties props = new Properties();
+        props.putAll(overrides);
+        props.putIfAbsent(ProducerConfig.ACKS_CONFIG, "-1");
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers());
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, keySerializer.getClass().getName());
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, valueSerializer.getClass().getName());
+        return new KafkaProducer<>(props);
+    }
+    
+    default <K, V> Producer<K, V> producer(Serializer<K> keySerializer, 
+                                           Serializer<V> valueSerializer
+    ) {
+        return producer(Collections.emptyMap(), keySerializer, valueSerializer);
+    }
+
+    default <K, V> Consumer<K, V> consumer(Map<String, Object> overrides,
+                                           Deserializer<K> keyDeserializer,
+                                           Deserializer<V> valueDeserializer
+    ) {
+        Properties props = new Properties();
+        props.putAll(overrides);
+        props.putIfAbsent(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.putIfAbsent(ConsumerConfig.GROUP_ID_CONFIG, "group");
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers());
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer.getClass().getName());
```

Comment:
maybe this helper method doesn't need to have `keyDeserializer`, as caller can cast the returned type manually.